src/Umbraco.Web.UI.Client/src/views/propertyeditors/mediapicker3/prevalue/mediapicker3.crops.controller.js

```diff
@@ -0,0 +1,110 @@
+angular.module("umbraco").controller("Umbraco.PropertyEditors.MediaPicker3.CropConfigurationController",
+	function ($scope) {
+
+        var unsubscribe = [];
+
+	    if (!$scope.model.value) {
+	        $scope.model.value = [];
+	    }
+
+        $scope.setFocus = false;
+
+	    $scope.remove = function (crop, evt) {
+	        evt.preventDefault();
+	        const i = $scope.model.value.indexOf(crop);
+            if (i > -1) {
+                $scope.model.value.splice(i, 1);
+            }
+	    };
+
+	    $scope.edit = function (crop, evt) {
+            evt.preventDefault();
+            crop.editMode = true;
+	    };
+
+	    $scope.addNewCrop = function (evt) {
+            evt.preventDefault();
+
+            var crop = {};
+            crop.editMode = true;
+
+            $scope.model.value.push(crop);
+            $scope.validate(crop);
+        }
+        $scope.setChanges = function (crop) {
+            $scope.validate(crop);
+            if(
+                crop.hasWidthError !== true &&
+                crop.hasHeightError !== true &&
+                crop.hasAliasError !== true
+            ) {
+                crop.editMode = false;
+                window.dispatchEvent(new Event('resize.umbImageGravity'));
```

Comment:
Why do we dispatch events on the window and not through the eventsService?

Reply:
It's a little hidden now that I changed the existing code from Jquery to native code. But the code in umbimagegravity.directive.js listens to this event on the window and I didn't want to change the existing code without a reason, so I just stuck to this. :-)

Reply:
All good then